README.md

```diff
@@ -0,0 +1,123 @@
+
+To train yolo_v3 algorithm to detect our custom objects we need to follow this steps:
+* Create a file named 'yolo-obj.cfg' as a configuration of the CNN (a custom copy of yolov3.cfg),
+* Create a file named 'obj.names' with the names of our custom object. Here is 'pneumonia',
+* Create a file named 'obj.data' with the numbers of class objects and path to different files,
+* Gather all images ('.jpg' format) in a same directory,
+* Create a '.txt' file with all labels in relative float value,
+* Create a '.txt' file with the path to all training images,
+* Create a '.txt' file with the path to all validation images,
+* Download the right [pre-trained weights file](https://github.com/AlexeyAB/darknet).
+
+To save model files more regularly during the training, we need to modify the code of this following file ```darknet/examples/detector.c``` (around the line 138) :
+```
+    if(i%10000==0 || (i < 1000 && i%100 == 0)){
+```
+to:
+```
+    if(i%1000==0 || (i < 2000 && i%50 == 0)){
+```
+We now save in the backup folder every 50 iterations a '.weights' file till we reach 2000 and then we save after every 1000 iterations
+
+
+## Global organization of the repository
+
+The folder **data** gather few files automatically generated by my algorithms.
+
+The file ```pneumonia_functions.py``` is mainly dedicated to prepare the algorithm to learn a new object. It offers the possibility to:
+* create the structure of the project and download automatically a pre trained weight given by the author of Yolo v3 ```def_structure```,
+* prepare different config files used by Yolo v3 to learn new objects ```def yolo_parameters```:
+  - ```def yolo_cfg_file```: generate the config file for yolo v3 training,
+  - ```def yolo_names_file```: name(s) of the new object(s),
+  - ```def yolo_data_file```: paths of all files needed for yolo v3 training.
+* set up the train et val dataset with ```def data_preprocessing```,
+* transform medical images (.dcm) into ".jpg" files with ```def yolo_jpg_file``` and ```def dcm_to_array```,
+* generate label in the shape of yolo_v3 learning CNN with ```def yolo_label_generation```. This function indicate to the algorithm where is located the object in the image training or image validation,
+* generate a '.txt' file with the path and the name of each image ```def yolo_image_path_file```,
+* display pneumonia or not image with or without the box ```def visualisation```,
+* represent the loss trend line over the learning process ```def loss_function```.
+
+The file ```pneumonia_detection.py```is dedicated to the detection on the test images. It allows to:
+* create a new '.cfg' file for yolo v3 detection as recommended by authors ```def test_cfg_file```,
+* launch detection on submitted image ```def detect```,
+* record all results under the right shape to be send to Kaggle ```def submission_file```.
+
+I offer the possibility to execute automatically all the algorithm through ```main.py```
+
```

Comment:
Yeah but here the targeted audience are people that might not go further than the README :(

Reply:
@Toz3wm The idea of a class is very interesting. To be honest, I don't feel at ease with class, but it will be a very good exercise. I add it to my "to do list".

@Ngoguey42 True, the targeted audience will be junior data scientist with no much knowledge in programming.